src/core/tsi/transport_security.cc

```diff
@@ -213,10 +213,10 @@ tsi_result tsi_handshaker_next(
 
 void tsi_handshaker_shutdown(tsi_handshaker* self) {
   if (self == nullptr || self->vtable == nullptr) return;
-  self->handshake_shutdown = true;
   if (self->vtable->shutdown != nullptr) {
     self->vtable->shutdown(self);
   }
+  self->handshake_shutdown = true;
```

Comment:
Why was this change needed?  What didn't work right in the previous position?

(I'm not objecting to this, just trying to understand.)

Reply:
Note that in `handshaker_shutdown()` of alts_tsi_handshaker.cc, since the flag `handshaker_shutdown` has already been set to `true` in the previous position, it will return directly without calling ` alts_handshaker_client_shutdown`.